src/axislike.rs

```diff
@@ -688,3 +705,105 @@ impl From<DualAxisData> for Vec2 {
         data.xy
     }
 }
+
+/// The shape of the deadzone for a [`DualAxis`] input.
+///
+/// Uses the x and y thresholds on [`DualAxis`] to set the shape diamensions
+#[derive(Debug, Clone, Copy, Serialize, Deserialize, PartialEq)]
+pub enum DeadZoneShape {
+    /// Deadzone with a cross
+    ///
+    /// Uses the x threshold as the horizontal width and y threshold as the vertical width.
+    Cross {
+        /// The width of the horizontal rectangle.
+        horizontal_width: f32,
+        /// The width of the vertical rectangle.
+        vertical_width: f32,
+    },
+    /// Deadzone with a rectangle.
+    ///
+    /// Uses the x threshold as the width and y threshold as the length.
+    Rect {
+        /// The width of the rectangle.
+        width: f32,
+        /// The height of the rectangle.
+        height: f32,
+    },
+    /// Deadzone with a circle.
+    ///
+    /// Uses the larger of the x or y threshold as the radius
+    Ellipse {
+        /// The horizontal radius
+        radius_x: f32,
+        /// The vertical radius
+        radius_y: f32,
+    },
+}
+
+impl Eq for DeadZoneShape {}
+impl std::hash::Hash for DeadZoneShape {
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {}
+}
+
+impl DeadZoneShape {
+    /// 
+    pub fn get_dual_axis_data(
+        &self,
+        dual_axis: &DualAxis,
+        input_stream: &InputStreams,
+    ) -> DualAxisData {
+        match self {
+            DeadZoneShape::Cross {
+                horizontal_width,
```

Comment:
Also, because `Cross` is more of a "snapzone", it makes `horizontal_width` and `vertical_width` harder to use as it requires setting the `SingleAxis` thresholds.